File: ros2_ws/src/my_py_pkg/my_py_pkg/smartphone.py
#!/usr/bin/env python3
#all these dependencies (imports of other ros packages) need to be added to the package.xml file
import rclpy
from rclpy.node import Node

from example_interfaces.msg import String 

class SmartPhoneNode(Node):
    def __init__(self):
        super().__init__("smartphone")
        #create subscriber
        self.subscriber_=self.create_subscription(msg_type=String, 
                                                  topic="robot_news_topic",
                                                  callback=self.callback_robot_news,   #each time the topic updates with a msg, the callback is executed
                                                  qos_profile=10)


        self.get_logger().info("Smartphone subscriber has started.")

    def callback_robot_news(self,msg):      #the msg is from the topic subscribed to 
        self.get_logger().info(msg.data)

def main(args=None):
    #first line to write in each ROS2 program. Initialize ROS2 communications
    rclpy.init(args=args)

    #create node 
    node=SmartPhoneNode()

    #Pause the program and allow the node to be alive
    rclpy.spin(node)

    #last line of program
    rclpy.shutdown()

if __name__=="__main_":
    main()

File: ros2_ws/src/my_cpp_pkg/src/smartphone.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"
class SmartPhoneNode : public rclcpp::Node //Modify Name
{
  public:
    SmartPhoneNode(): Node("smartphone")  //Modify Name
    {
        //Initialize the subscriber with the topic its subscribed to and the 
        //callback called when the topic updates
        subscriber_=this->create_subscription<example_interfaces::msg::String>(
            "robot_news_topic",
            10,
            std::bind(&SmartPhoneNode::readNews,this,std::placeholders::_1));
        RCLCPP_INFO(this->get_logger()," Smartphone node has started!");


    };

   private:
   rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber_;

   //Callback for reading msgs from topics
   void readNews(const example_interfaces::msg::String::SharedPtr message)
   {

       RCLCPP_INFO(this->get_logger(),message->data.c_str());
   };
};

int main(int argc, char** argv)
{
    //Initialise the node
    rclcpp::init(argc,argv);

    //Create the node
    auto node = std::make_shared<SmartPhoneNode>();
    
    //Keep the node alive
    rclcpp::spin(node);

    //Shutdown node
    rclcpp::shutdown();
    return 0;

}